src/include/gurt/common.h

```diff
@@ -322,8 +322,40 @@ int d_rank_list_append(d_rank_list_t *rank_list, d_rank_t rank);
 int d_rank_list_dump(d_rank_list_t *rank_list, d_string_t name, int name_len);
 d_rank_list_t *uint32_array_to_rank_list(uint32_t *ints, size_t len);
 int rank_list_to_uint32_array(d_rank_list_t *rl, uint32_t **ints, size_t *len);
-int d_sgl_init(d_sg_list_t *sgl, unsigned int nr);
-void d_sgl_fini(d_sg_list_t *sgl, bool free_iovs);
+
+static inline int
+d_sgl_init(d_sg_list_t *sgl, unsigned int nr)
+{
+	sgl->sg_nr_out = 0;
+	sgl->sg_nr = nr;
+
+	if (unlikely(nr == 0)) {
```

Comment:
If this is valid then we probably don't want an unlikely here.

Reply:
This is just for hint as very low probability. I don't think we have this case be frequently.